pkg/controller/certificatemanager/certificateauthority/controller.go

```diff
@@ -0,0 +1,280 @@
+			managed.WithConnectionPublishers(),
+			managed.WithInitializers(),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	client      client.Client
+	newClientFn func(*aws.Config) (acmpca.Client, error)
+	awsConfigFn func(context.Context, client.Reader, *corev1.ObjectReference) (*aws.Config, error)
+}
+
+func (conn *connector) Connect(ctx context.Context, mgd resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mgd.(*v1alpha1.CertificateAuthority)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+
+	awsconfig, err := conn.awsConfigFn(ctx, conn.client, cr.Spec.ProviderReference)
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := conn.newClientFn(awsconfig)
+	if err != nil {
+		return nil, errors.Wrap(err, errClient)
+	}
+	return &external{c, conn.client}, nil
+}
+
+type external struct {
+	client acmpca.Client
+	kube   client.Client
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	cr, ok := mgd.(*v1alpha1.CertificateAuthority)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	response, err := e.client.DescribeCertificateAuthorityRequest(&awsacmpca.DescribeCertificateAuthorityInput{
+		CertificateAuthorityArn: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(acmpca.IsErrorNotFound, err), errGet)
+	}
+
+	if response.CertificateAuthority == nil {
+		return managed.ExternalObservation{}, errors.New(errEmpty)
+	}
+
+	certificateAuthority := *response.CertificateAuthority
+	current := cr.Spec.ForProvider.DeepCopy()
+	acmpca.LateInitializeCertificateAuthority(&cr.Spec.ForProvider, &certificateAuthority)
+
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+	cr.SetConditions(runtimev1alpha1.Available())
+
+	cr.Status.AtProvider = acmpca.GenerateCertificateAuthorityExternalStatus(certificateAuthority)
+
+	tags, err := e.client.ListTagsRequest(&awsacmpca.ListTagsInput{
+		CertificateAuthorityArn: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(acmpca.IsErrorNotFound, err), errListTagsFailed)
+	}
+
+	upToDate := acmpca.IsCertificateAuthorityUpToDate(cr, certificateAuthority, tags.Tags)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errUpToDateFailed)
+	}
+
+	// Check the PCA status and return error if PCA is in Pending State.
```

Comment:
Returning an error is going to cause immediate requeue. I don't think we want to do that here right?

Reply:
Got it,
added check in update method so that it will throw an error when PCA is in Pending status.